src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLogger.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class LinuxAppServiceFileLogger
+    {
+        private readonly string _logFileName;
+        private readonly string _logFileDirectory;
+        private readonly string _logFilePath;
+        private readonly BlockingCollection<string> _buffer;
+        private readonly List<string> _currentBatch;
+        private readonly ILinuxAppServiceFileSystem _fileSystem;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private Task _outputTask;
+
+        public LinuxAppServiceFileLogger(string logFileName, string logFileDirectory, ILinuxAppServiceFileSystem fileSystem, bool startOnCreate = true)
+        {
+            _logFileName = logFileName;
+            _logFileDirectory = logFileDirectory;
+            _logFilePath = Path.Combine(_logFileDirectory, _logFileName + ".log");
+            _buffer = new BlockingCollection<string>(new ConcurrentQueue<string>());
+            _currentBatch = new List<string>();
+            _fileSystem = fileSystem;
+            _cancellationTokenSource = new CancellationTokenSource();
+
+            if (startOnCreate)
+            {
+                Start();
+            }
+        }
+
+        // Maximum number of files
+        public int MaxFileCount { get; set; } = 3;
```

Comment:
Note we already have a FileWriter that we use for our other file system logs. It does auto purge, throttling, etc. Have you seen that? Would be good for us to not have to maintain 2 of these.

Reply:
I saw that. but that class would need some changes to the filename format and few other things. so i thought it was better to keep this separate.